src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -352,6 +353,45 @@ public function getUploadedFilesAction(int $id)
         return array($fileInfo);
     }
 
+    /**
+     * Return a list of files uploaded for a dataset submission.
+     *
+     * @param integer $id The id of the dataset submission.
+     *
+     * @Route(
+     *     "/api/files_dataset_submission/{id}",
+     *     name="pelagos_api_get_files_dataset_submission",
+     *     methods={"GET"},
+     *     defaults={"_format"="json"},
+     *     requirements={"id"="\d+"}
+     *     )
+     *
+     * @View()
+     *
+     * @return array The list of uploaded files.
+     */
+    public function getFiles(int $id)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        $fileData = array();
+        $datasetSubmission = $this->handleGetOne(DatasetSubmission::class, $id);
```

Comment:
Don't use entityHandler anymore, use a/the repository instead.

Reply:
This whole block was just moved over from the other File, didn't add it in this pull request, we can do it but i feel it shouldn't be part of this PR